ceremonies/src/lib.rs

```diff
@@ -944,6 +951,42 @@ impl<T: Config> Pallet<T> {
 		Ok(participant_type)
 	}
 
+	/// removes a participant from the registry maintaining continuous indices
+	fn unregister_newbie(
+		cid: CommunityIdentifier,
+		cindex: CeremonyIndexType,
+		participant: &T::AccountId,
+	) -> Result<(), Error<T>> {
+		if <encointer_scheduler::Pallet<T>>::current_phase() != CeremonyPhaseType::REGISTERING {
+			return Err(<Error<T>>::WrongPhaseForUnregistering.into())
+		}
+
+		if !<NewbieIndex<T>>::contains_key((cid, cindex), &participant) {
+			return Ok(())
+		}
+
+		let participant_index = <NewbieIndex<T>>::get((cid, cindex), &participant);
+		let participant_count = <NewbieCount<T>>::get((cid, cindex));
+
+		let last_participant = <NewbieRegistry<T>>::get((cid, cindex), &participant_count)
+			.expect("Indices are continuous, thus index participant_count is Some; qed");
+
+		<NewbieRegistry<T>>::insert((cid, cindex), &participant_index, &last_participant);
+		<NewbieIndex<T>>::insert((cid, cindex), &last_participant, &participant_index);
+
+		<NewbieRegistry<T>>::remove((cid, cindex), &participant_count);
+		<NewbieIndex<T>>::remove((cid, cindex), &participant);
```

Comment:
I think you are manually implementing the swap method I recommended in the comments before. Is there a reason for that?

Reply:
Ah yes, you are right, you actually make it better than what I proposed XD